### ISSUE TITLE
Do we need to deal with correlated variables?

### ISSUE THREAD

**Issue by gideonite** [ai2cm/ace#74]
In classical ML / stats, correlated variables are considered bad. They mess with regularization / weight-decay. What about for our models? What sorts of inductive biases can we use to group variables together for the model?

**Comment by gideonite** [comment]
This is probably obvious to the NWP physicists, but perhaps its worth making a few plots to illustrate this. Qualitatively looking at the figures below, it would seem that correlation of features ⇒ correlations of errors.

Here are correlations of the incoming data variables:

![Image](https://user-images.githubusercontent.com/2163686/232164430-d10f0c0b-bd57-4ec2-b01c-595853beba39.png)

Here are correlations of errors evolving over time, of the rollouts at inference-time.

![Image](https://user-images.githubusercontent.com/2163686/232164595-12a9b84e-c053-4612-b5e4-f265d65dd3d8.png)

Notebooks:
https://github.com/ai2cm/explore/blob/master/gideond/full-model-emulation/1.0%20Decompose%20Errors%20By%20Variable.ipynb

https://github.com/ai2cm/explore/blob/master/gideond/full-model-emulation/0.0%20Exploring%20Correlation%20Structure.ipynb

**Comment by mcgibbon** [comment]
I would be cautious about putting too much work into this up-front without seeing particular issues. If this is causing skill or stability issues, we should come up with some measure of improvement and ensure that any approaches related to this improve the model.

A danger with applying "fixes" for this is you may destroy physically-meaningful information and degrade model performance. For example, if you decompose the vertical into principal components and use those as the input features, the model loses the ability to perform vertically-local operations (or at least, these become much more complicated to perform).

Correlated issues in and of themselves are not a problem - state variables are correlated, and we expect the "error" from a perfect emulator model with a perturbed initial condition to be correlated across variables as it evolves in time.

In short, I'd first ask "do we need to deal with correlated variables?".

**Comment by gideonite** [comment]
> In short, I'd first ask "do we need to deal with correlated variables?"

Done

**Comment by gideonite** [comment]
Following up on our conversation today. There are a number of ways to tackle this (below). For now, it's fine to be aware of this.

1. Remove variables
2. Transform the variables / feature engineering -- vertical integration, other physically meaningful transformations
3. Modify the model -- parameterize vertical integration using convolutions, tune regularization / weight-decay.

**Comment by gideonite** [comment]
Sounds like the answer is no. This plot does not control for spatial or temporal (e.g. seasonal, diurnal, etc.) correlations and given that we are unlikely to learn something that we don't already know from further investigation into this, it's worth closing for now.
